eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -0,0 +1,258 @@
+    $groupId = $pkg.GroupId
+  }
+
+  if ($groupId) {
+    $pkgKey = "${groupId}:${pkgKey}"
+  }
+
+  return $pkgKey
+}
+
+# TODO: Move this function elsewhere in eng/common:
+function GetPackageLookup($packageList) {
+  $packageLookup = @{}
+
+  foreach ($pkg in $packageList) {
+    $pkgKey = GetPackageKey $pkg
+
+    # We want to prefer updating non-hidden packages but if there is only
+    # a hidden entry then we will return that
+    if (!$packageLookup.ContainsKey($pkgKey) -or $packageLookup[$pkgKey].Hide -eq "true") {
+      $packageLookup[$pkgKey] = $pkg
+    }
+    else {
+      # Warn if there are more then one non-hidden package
+      if ($pkg.Hide -ne "true") {
+        Write-Host "Found more than one package entry for $($pkg.Package) selecting the first non-hidden one."
+      }
+    }
+
+    if ($pkg.PSObject.Members.Name -contains "GroupId" -and ($pkg.New -eq "true") -and $pkg.Package) {
+      $pkgKey = $pkg.Package
+      if (!$packageLookup.ContainsKey($pkgKey)) {
+        $packageLookup[$pkgKey] = $pkg
+      }
+      else {
+        $packageValue = $packageLookup[$pkgKey]
+        Write-Host "Found more than one package entry for $($packageValue.Package) selecting the first one with groupId $($packageValue.GroupId), skipping $($pkg.GroupId)"
+      }
+    }
+  }
+
+  return $packageLookup
+}
+
+$onboardedPackages = &$GetOnboardedDocsMsPackagesFn `
+  -DocRepoLocation $DocRepoLocation
+
+# This criteria is different from criteria used in `Update-DocsMsPackages.ps1`
+# because we need to generate ToCs for packages which are not necessarily "New"
+# in the metadata AND onboard legacy packages (which `Update-DocsMsPackages.ps1`
+# does not do)
+$metadata = (Get-CSVMetadata).Where({
+    $_.Package `
+      -and $onboardedPackages.ContainsKey($_.Package) `
+      -and $_.Hide -ne 'true'
+  })
+
+$fileMetadata = @()
+foreach ($metadataFile in Get-ChildItem "$DocRepoLocation/metadata/*/*.json" -Recurse) {
+  $fileContent = Get-Content $metadataFile -Raw
+  $metadataEntry = ConvertFrom-Json $fileContent
+
+  if ($metadataEntry) {
+    $fileMetadata += $metadataEntry
+  }
+}
+
+# Add file metadata information to package metadata from metadata CSV. Because
+# metadata can exist for packages in both preview and GA there may be more than
+# one file metadata entry. If that is the case keep the first entry found. We
+# only use the `DirectoryPath` property from the json file metadata at this time
+for ($i = 0; $i -lt $metadata.Count; $i++) {
+  foreach ($fileEntry in $fileMetadata) {
+    if ($fileEntry.Name -eq $metadata[$i].Package) {
+      if ($metadata[$i].PSObject.Members.Name -contains "FileMetadata") {
+        Write-Host "File metadata already added for $($metadata[$i].Package). Keeping the first entry found."
+    continue
+  }
+
+      Add-Member `
+        -InputObject $metadata[$i] `
+        -MemberType NoteProperty `
+        -Name FileMetadata `
+        -Value $fileEntry
+  }
+  }
```

Comment:
The indentation is a little wonky here and makes it look like you have a trailing curly brace (though I think it rather belongs to the if statements).

Reply:
Interesting. This is my first time using the vscode source control extension and it appears to not be showing me whitespace changes to stage.